Libraries/JavaScriptAppEngine/System/JSTimers/JSTimers.js

```diff
@@ -106,10 +106,12 @@ var JSTimers = {
 
   clearImmediate: function(timerID) {
     JSTimers._clearTimerID(timerID);
-    JSTimersExecution.immediates.splice(
-      JSTimersExecution.immediates.indexOf(timerID),
-      1
-    );
+    if (timerID != null) {
```

Comment:
We want `!=` here so it works will `null` and `undefined`. It's the same check the is used in other places in this file.